### DISCUSSION TITLE
Invoking Directus cloud API from localhost causes CORS error

### DISCUSSION THREAD

**Discussion by j-petty** [directus/directus#15518] [Archive]
Using a Directus cloud project when running my front-end application locally and invoking the Directus API it returns a CORS error:

> Access to XMLHttpRequest at 'http://{project}.directus.app/items/{collection}' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.

Is there a way to configure CORS settings for a Directus cloud project? When running a self-hosted application it's possible using the [configuration options](https://docs.directus.io/self-hosted/config-options.html#cors).

**Comment by azrikahar** [comment]
It might be because you're calling the cloud project's API with `http` instead of `https`. Can you try changing it to `https` to see whether that resolves it?

**Reply by j-petty** [reply]
Spot on! I checked back through our config and we were inadvertently calling the API via HTTPS because our config was using `//{project}.directus.app` meaning requests running on localhost went to HTTP. Should have noticed that in the exception above.. Good pickup!
